### DISCUSSION TITLE
The January meeting and getting towards v1.0

### DISCUSSION THREAD

**Discussion by webmiche** [xdslproject/xdsl#277] [Ideas]
I decided to note a couple of ideas here of what we can/could/should do in January. Sometime in the future, this can be converted to separate issues or a github project or whatever.

To me, it feels like xDSL is starting to go towards its v1.0 release. The fundamental data structures are there and, with the xDSL paper being submitted, we will (hopefully) get a big stage next summer. It would be really cool if we could say that xDSLv1.0 has been released in this presentation and is ready for use.

I feel the January meeting should be the start of the process towards that. Now, what is there left todo. IMO, there are a couple of things:
- The pyrdl-MLIR conversion (back and forth) should be there.
- The documentation and tutorials need to be overhauled.
- Our code coverage should reach as close to 100% as possible.
- The rewrite engine needs more love/be developed further.

I feel that it would be cool if we could addresse/solve the first three in this meeting. The exact action points can be developed further, closer to the meeting.

Also, towards a v1.0 release means that there will be more light on xDSL as a whole. Therefore, I feel that we want to be sure about the design of data strutures and everything as there will be higher cost related to this after the 1.0 release.

**Comment by math-fehr** [comment]
Yes, I would also add:
* Remake of the syntax, and probably taking the MLIR syntax (even if it is a bit ugly)
* Ensuring we check types with PyRight, and also probably stop using yapf and using something better.
* Maybe moving the dialects to another repo?

**Comment by webmiche** [comment]
From #226: unify (and document) constructor names

**Comment by superlopuh** [comment]
I vote never! In any case, I think we can safely close this issue and have more issues for stabilising parts of the framework's API even if I'm not sure that it's actually possible to stabilise the API throughout the framework.
